### PR TITLE
imported typeahead from npm and upgraded it

### PR DESCRIPTION
**Pull request by vitoravelino**
Initially I kept the same version but I took the liberty to upgrade it to the newer one because it exports the object properly instead of just exposing it as a global variable.

The behavior still the same but if the project prefers to keep the previous version, just let me know and I'll downgrade it.

Signed-off-by: Vítor Avelino <contact@vitoravelino.me>

### CODE REVIEW

File: app/assets/stylesheets/components/typeahead.scss [hunk 1]
```diff
@@ -1,7 +1,46 @@
+.tt-menu {
```
Comment (mssola) [inline app/assets/stylesheets/components/typeahead.scss]: This file should be kept in vendor I think, since we took it from the gem iirc.

Comment (vitoravelino) [inline app/assets/stylesheets/components/typeahead.scss]: You are right. I made 2 or 3 changes due upgrade and previous enhancement but I'll keep it in vendor folder. Thanks for reminding me that. :)

**Comment by vitoravelino** [comment]
Done!

**Comment by mssola** [comment]
Thanks 👏

**Pull request merged**